HEN_HOUSE/egs++/egs_object_factory.cpp

```diff
@@ -190,6 +190,7 @@ EGS_Object *EGS_ObjectFactory::createObjects(EGS_Input *i,
         if (!o) egsWarning("EGS_ObjectFactory::createObjects(): an object "
                                "with the name %s does not exist\n",sought_object.c_str());
     }
+    delete input;
```

Comment:
`input` is allocated here by `takeInputItem` but never deallocated. 
https://github.com/nrc-cnrc/EGSnrc/blob/3e0db683e48a328b8fede9e7756e0deb693fbf9e/HEN_HOUSE/egs%2B%2B/egs_object_factory.cpp#L162-L163
https://github.com/nrc-cnrc/EGSnrc/blob/3e0db683e48a328b8fede9e7756e0deb693fbf9e/HEN_HOUSE/egs%2B%2B/egs_input.cpp#L226-L237